Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/HoughTransformSeeder.hpp

```diff
@@ -0,0 +1,267 @@
+/// like a convolution. Also stored are indices of all hits that contributed to
+/// each bin. Size m_imageSize_y * m_imageSize_x. (NOTE y is row coordinate) For
+/// now, what is stored is actually the index of the object in the vectors, so
+/// we can get the Index layer
+
+namespace Acts {
+class TrackingGeometry;
+}
+
+namespace ActsExamples {
+/// Used in multiple places. The 2d vector refers to the 2d image. For a single
+/// layer, the int refers to the number of hits in the bin of the image
+//// For the total image, the int counts the number of layers with one or more
+/// hit in that bin
+// The unsigned is a counter that will point to a spacepoint or to a measurement
+// object
+typedef vector2D<std::pair<int, std::unordered_set<unsigned>>> Image;
+
+/// Construct track seeds from space points.
+class HoughTransformSeeder final : public BareAlgorithm {
+ public:
+  struct Config {
+    /// Input space point collections.
+    ///
+    /// We allow multiple space point collections to allow different parts of
+    /// the detector to use different algorithms for space point construction,
+    /// e.g. single-hit space points for pixel-like detectors or double-hit
+    /// space points for strip-like detectors.
+    /// Note that we don't *need* spacepoints (measurements can be used instead)
+    std::vector<std::string> inputSpacePoints;
+    /// Output track seed collection.
+    std::string outputSeeds;
+    /// Output hough track collection.
+    std::string outputProtoTracks;
+    /// Input source links collection.
+    std::string inputSourceLinks;
+    std::shared_ptr<const Acts::TrackingGeometry> trackingGeometry;
+    /// For which part of the detector geometry should space points be created.
+    ///
+    /// Only volumes and layers can be set. Zero values can be used as wildcards
+    /// to select larger parts of the hierarchy, i.e. setting only the volume
+    /// selects all measurements within that volume. Adding a single identifier
+    /// with all components set to zero selects all available measurements. The
+    /// selection must not have duplicates.
+    std::vector<Acts::GeometryIdentifier> geometrySelection;
+
+    /// Input measurements collection.
+    std::string inputMeasurements;
+
+    std::vector<int> m_subRegions = {
+        -1};  // -1 for entire region (no slicing), but this can be more than
+              // one region if data are sliced
+
+    unsigned m_nLayers = 10;  // total number of layers
+
+    /// Trace each hit that goes in a bin
+    /// Disabling this will save memory/time since each bin doesn't have to
+    /// store all its hits but the roads created won't have hits from
+    /// convolution, etc.
+    bool m_traceHits = true;
+
+    float m_xMin = 0;            // minphi
+    float m_xMax = 2 * 3.14159;  // maxphi
+    float m_yMin = -1.0;         // min q/pt, -1/1 GeV
+    float m_yMax = 1.0;          // max q/pt, +1/1 GeV
```

Comment:
Was sort of hoping to try and enforce the idea that this could be implemented in other ways (we have an eta-z0 HT that has been worked on in EFTracking, for example). I'm not set on that if you think it really is important, of course